validator/client/iface/validator.go

```diff
@@ -57,4 +57,6 @@ type Validator interface {
 	ReceiveBlocks(ctx context.Context, connectionErrorChannel chan<- error)
 	HandleKeyReload(ctx context.Context, newKeys [][fieldparams.BLSPubkeyLength]byte) (bool, error)
 	CheckDoppelGanger(ctx context.Context) error
+	PrepareBeaconProposer(ctx context.Context, km keymanager.IKeymanager) error
+	SetPubKeyToValidatorIndexMap(ctx context.Context, km keymanager.IKeymanager) error
```

Comment:
For some reason this doesnt feel like something that should be exposed in an interface...as it feels like an implementation detail. Perhaps it can be renamed or put inside a specific method instead where it is used

Reply:
removed it